packages/connection-storage/src/connection-storage.ts

```diff
@@ -229,6 +238,7 @@ export class ConnectionStorage {
       );
     }
     const connectionInfo = mergeSecrets(storedConnectionInfo!, secrets);
+    this.migrateSavedConnectionType(connectionInfo);
```

Comment:
[tiny nit] We mostly try to use pure functions here that return a new `connectionInfo`, I'd maybe just stick to that pattern